fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/ResolverValidator.java

```diff
@@ -0,0 +1,301 @@
+                            new ScanProperties(executeProperties.build())),
+                3)) {
+            resolver.getDatabase().asyncToSync(timer, FDBStoreTimer.Waits.WAIT_VALIDATE_RESOLVER,
+                    cursor.forEach(validatedEntry -> {
+                        if (!badEntriesOnly || validatedEntry.getValidationResult() != ValidationResult.OK) {
+                            entryListener.accept(validatedEntry);
+                        }
+                    }));
+        }
+    }
+
+    /**
+     * Validate entries in a {@link LocatableResolver}'s mappings. This method scans the forward mapping
+     * ({@code String} -&gt; {@code long}) and validates that the entry has a corresponding entry in the
+     * reverse mapping, as well as validating that the entry in the reverse mapping points to the forward
+     * mapping key.
+     *
+     * @param resolver the resolver to scan
+     * @param context the transaction context to use
+     * @param continuation continuation from a previous scan
+     * @param repairMissingEntries if {@code true}, missing reverse directory entries will be restored
+     * @param scanProperties scan properties that control the nature of the scan
+     * @return a cursor returning the disposition of all scanned entries in the resolvers mapping table
+     */
+    public static RecordCursor<ValidatedEntry> validate(@Nonnull final LocatableResolver resolver,
+                                                        @Nonnull final FDBRecordContext context,
+                                                        @Nullable final byte[] continuation,
+                                                        final boolean repairMissingEntries,
+                                                        @Nonnull final ScanProperties scanProperties) {
+        return validate(resolver, context, continuation, 10, repairMissingEntries, scanProperties);
+    }
+
+    /**
+     * Validate entries in a {@link LocatableResolver}'s mappings. This method scans the forward mapping
+     * ({@code String} -&gt; {@code long}) and validates that the entry has a corresponding entry in the
+     * reverse mapping, as well as validating that the entry in the reverse mapping points to the forward
+     * mapping key. For example, if {@code "foo" -&gt; 10} is in the forward mapping and {@code "foo"} is
+     * not found in the reverse mapping, it will be returned as a bad entry. Similarly, if {@code "foo"} is
+     * found in the reverse mapping, but it has a value of {@code 30}, then that is considered an error.
+     *
+     * @param resolver the resolver to scan
+     * @param context the transaction context to use
+     * @param continuation continuation from a previous scan
+     * @param reverseLookupPipelineSize the number of reverse directory lookups to perform in parallel while
+     *   scanning the forward mappings
+     * @param repairMissingEntries if {@code true}, missing reverse directory entries will be restored
```

Comment:
It's probably worth noting somewhere that it's still the caller's responsibility to commit the context to actually repair any missing entries (if I'm reading everything correctly here). I assume that the difficulty of hooking that into the `AutoContinuingCursor` is partially why the synchronous "validate in multiple transactions" method doesn't have a `repairMissingEntries` parameter?